### PR TITLE
Doxygen update for virtio structures

### PR DESCRIPTION
**Pull request by arnopo**
Complete virtio structure doxygen description initiate in https://github.com/OpenAMP/open-amp/pull/507

### CODE REVIEW

File: lib/include/openamp/virtio.h [hunk 1]
```diff
@@ -217,41 +227,73 @@ struct virtio_device {
 /*
  * Helper functions.
  */
+
+/**
+ * @brief Get the name of a virtio device.
+ *
+ * @param devid Id of the device.
```
Comment (tammyleino) [inline lib/include/openamp/virtio.h:234]: Maybe just devid? or dev id?

File: lib/include/openamp/virtqueue.h [hunk 2]
```diff
@@ -63,66 +63,98 @@ extern "C" {
...
-	/*
-	 * Last consumed descriptor in the used table,
-	 * trails vq_ring.used->idx.
+	/**
+	 * Head of the free chain in the descriptor table. If there are no free descriptors,
+	 * this will be set to VQ_RING_DESC_CHAIN_END.
 	 */
+	uint16_t vq_desc_head_idx;
+
+	/* Last consumed descriptor in the used table, trails vq_ring.used->idx. */
```
Comment (tammyleino) [inline lib/include/openamp/virtqueue.h:128]: Missing /**

File: lib/include/openamp/virtio.h [hunk 3]
```diff
@@ -217,41 +227,73 @@ struct virtio_device {
...
 	void (*read_config)(struct virtio_device *dev, uint32_t offset,
 			    void *dst, int length);
+
+	/**
+	 * Write a variable amount from the device specific (ie, network)
+	 * configuration region.
+	 */
 	void (*write_config)(struct virtio_device *dev, uint32_t offset,
 			     void *src, int length);
+	/** Request a reset of the virtio device. */
```
Comment (tammyleino) [inline lib/include/openamp/virtio.h:293]: Add blank line above for consistency?

**Comment by tammyleino** [comment]
Added a few minor comments, but looks good otherwise.  Thanks!

File: lib/include/openamp/virtio.h (hunk 3 above)
Comment (arnopo) [reply lib/include/openamp/virtio.h:294]: done

File: lib/include/openamp/virtqueue.h (hunk 2 above)
Comment (arnopo) [reply lib/include/openamp/virtqueue.h]: done

File: lib/include/openamp/virtqueue.h [hunk 4]
```diff
@@ -63,66 +63,98 @@ extern "C" {
...
 	uint16_t vq_used_cons_idx;
 
-	/*
-	 * Last consumed descriptor in the available table -
-	 * used by the consumer side.
-	 */
+	/* Last consumed descriptor in the available table, used by the consumer side. */
```
[the remark this answers is not in the document]
Comment (arnopo) [reply lib/include/openamp/virtqueue.h]: done

**Review by tammyleino — APPROVED** [review]

File: lib/include/openamp/virtio_ring.h [hunk 5]
```diff
@@ -162,6 +148,19 @@ struct vring {
  *
  * NOTE: for VirtIO PCI, align is 4096.
  */
+struct vring {
+	/** The number of descriptors in the virtqueue */
```
Comment (tnmysh) [inline lib/include/openamp/virtio_ring.h:152]: We can add in doxygen document that num is expected to be power of 2.

**Review by tnmysh — APPROVED** [review]
Approved with minor comment that can be fixed later as well.

**Pull request merged**